### ISSUE TITLE
Vyřešit vícejazyčnost pro dílčí API

### ISSUE THREAD

**Issue by motyc** [ARUP-CAS/arup-da-amcr#88]
Bylo by dobré umět podávat výsledky v obou jazykových verzích, ať už rozlišením pomocí atributu (lepší pokud půjde), nebo na základě parametru API (horší, ale také schůdné). RDF dělám rovnou v CS i EN, ale je do toho třeba nějak zapracovat překlady heslářových položek.

**Comment by vbar** [comment]
Tohle bych řešil standardně přes HTTP Accept-Language - Tomcat podle něj nastavuje lokalizaci, a generování výstupního XML by ji mělo brát v potaz. Na vstupu ovšem musí mít oba jazyky, tj. RDF (a jeho zdroj exportované XML) by měli být v CS i EN, včetně heslářů...

**Comment by motyc** [comment]
Nebál bych se zachovat oba jazyky i na výstupu - RDF to standardně umožňuje a ani jinde by to asi nebyl problém. Háček je v tom vstupu, protože momentálně se překlady řeší klíčováním heslářů ze souborů jen pro Digiarchiv. Teoreticky by se vybrané položky XML při generování mohly automaticky přeložit stejným principem a doplnit do XML jako druhá jazyková verze tagu, ale je ke zvážení, jak to dělat.

Samozřejmě nejlepší by bylo, zrušit stávající CSV s překlady a zahrnout angličtinu přímo do heslářů. Pak by to bylo více než snadné. Skoro bych to preferoval, i kdyby se to mělo řešit jako samostatný úkol. Asi bych rozhodnutí ještě krapet odložil a až vyřešíme základ, můžeme se dohodnout na nejschůdnější cestě.

**Comment by vbar** [comment]
Ano, ve výstupu RDF můžeme klidně zachovat oba jazyky. Na druhé straně v Dublin Core bych se držel jednoho - tam si nejsem jist jsou-li vícejazyčná data vůbec podporovaná, a určitě nejsou obvyklá... Jak dostat více jazyků do RDF uvidíme...

**Comment by motyc** [comment]
Na tuto věc jsme při schůzce zapomněli - je realistické vícejazyčnost implementovat nyní na základě existujících překladových souborů pro hesláře, které se používají pro digitální archiv? Takto to původně bylo myšleno... Nejde ani tak o to, jestli se to stihne časově, ale jestli je to ještě proveditelné v rámci dojednaného rozpočtu. Znamenalo by mít možnost zvolit zda výsledky chci v cs nebo en. Na základě toho by se naklíčovala heslářová pole (jejich výčet a napojení na dílčí hesláře mohu v podstatě obratem dodat).

**Comment by vbar** [comment]
Překladové soubory pro hesláře mi nepřipadají jako nejvhodnější formát pro překlad XML - v první řadě se k nim nové API nedostane (a když je zkopírujeme, tak se nám rozjedou). Mnohem raději bych měl překlady v databázi, a rozšířil exportované XML o jazykové mutace...

**Comment by motyc** [comment]
Chápu a je to asi opravdu vhodnější. Dobře, nechme to tedy prozatím plavat, jako případné vylepšení do budoucna. Asi nemá smysl se teď trápit něčím, co bychom stejně předělávali.
